### DISCUSSION TITLE
How would i make it so when pause the game and quit instead of reseting the game it takes me back to a scene

### DISCUSSION THREAD

**Discussion by Raptorq1** [Techokami/SonicWorldsNext#163] [Q&A] [answered]
How would i make it so when pause the game and quit instead of reseting the game it takes me back to a scene
[PauseManager2.zip](https://github.com/user-attachments/files/15905149/PauseManager2.zip)

**Comment by r6915ee** [comment]
Replace `line 148` in your script with this line of code:

`Global.main.change_scene_to_file(preload("res://Scene/Presentation/Title.tscn"))`

I am only using `Title.tscn` as an example, so put in whatever scene you need in place of it. Preloading will likely be necessary, because in its original state, the line of code was `Global.main.change_scene_to_file(Global.startScene)`; `startScene` is a variable that preloads `Title.tscn` by default.

**Reply by Raptorq1** [reply]
Okay but the the pause menu still shows

**Reply by r6915ee** [reply]
What would happen originally if you hadn't changed the line?

**Reply by Raptorq1** [reply]
Resets the game

**Reply by r6915ee** [reply]
What definition do you mean? Does it just send you to the title screen, or does the game itself close and reopen?

**Reply by Raptorq1** [reply]
Send me to the title screen

**Reply by r6915ee** [reply]
Did the pause menu still show up?

**Reply by Raptorq1** [reply]
I am having little problem with the code when I back out to the menu and reenter a zone it makes the pause menu to stop working

**Reply by r6915ee** [reply]
Does the game output anything specific in the event that this occurs?

**Reply by Raptorq1** [reply]
No oh and here's the code

await get_tree().process_frame
						visible = false
						set_menu(0)
						get_tree().paused = false
						Global.effectTheme.stop()
						Global.bossMusic.stop()
						Global.main.change_scene_to_file(preload("res://Data/Stages/Menu/menu.tscn"))

**Reply by r6915ee** [reply]
Does the game still pause, just without the pause menu appearing?

**Reply by Raptorq1** [reply]
No it doesn't pause

**Reply by r6915ee** [reply, marked as answer]
You might need to include a line like this in that moment:

```gdscript
Global.main.wasPaused = false
```

The pause menu to my knowledge directly checks that boolean. Restarting the level also sets it to `false` generally, so that variable might be causing the problem because it still thinks that it should be paused, regardless of the actual state of the pause menu.

**Comment by NeonSRB2** [comment]
> How would i make it so when pause the game and quit instead of reseting the game it takes me back to a scene [PauseManager2.zip](https://github.com/user-attachments/files/15905149/PauseManager2.zip)

It only resets the game because you're testing, in the actual game it would just close. You might want to include a replacement somewhere around the title screen or main menu if you add one.
